### ISSUE TITLE
Could not handle BIGNUMERICs while reading

### ISSUE THREAD

**Issue by devendrasr** [GoogleCloudDataproc/spark-bigquery-connector#289]
We are unable to read from bigquery getting below exception while reading BIGNUMERIC data types:  

```
Exception in thread "main" java.lang.IllegalStateException: Unexpected type: BIGNUMERIC
	at com.google.cloud.spark.bigquery.SchemaConverters.getStandardDataType(SchemaConverters.java:234)
	at com.google.cloud.spark.bigquery.SchemaConverters.lambda$getDataType$4(SchemaConverters.java:184)
	at java.util.Optional.orElseGet(Optional.java:267)
	at com.google.cloud.spark.bigquery.SchemaConverters.getDataType(SchemaConverters.java:184)
	at com.google.cloud.spark.bigquery.SchemaConverters.convert(SchemaConverters.java:166)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at com.google.cloud.spark.bigquery.SchemaConverters.toSpark(SchemaConverters.java:54)
	at com.google.cloud.spark.bigquery.BigQueryRelation.<init>(BigQueryRelation.scala:37)
	at com.google.cloud.spark.bigquery.direct.DirectBigQueryRelation.<init>(DirectBigQueryRelation.scala:48)
	at com.google.cloud.spark.bigquery.BigQueryRelationProvider.createRelationInternal(BigQueryRelationProvider.scala:78)
	at com.google.cloud.spark.bigquery.BigQueryRelationProvider.createRelation(BigQueryRelationProvider.scala:45)
	at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:318)
	at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:223)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
	at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:167)```

**Comment by devendrasr** [comment]
I am wondering this fix is still missing in the latest release! +1

**Comment by richard-williamson** [comment]
@davidrabinowitz  - is it possible to get someone working on this?

**Comment by maasg** [comment]
+1 - I just hit the same
```
java.lang.IllegalStateException: Unexpected type: BIGNUMERIC 	at com.google.cloud.spark.bigquery.SchemaConverters.getStandardDataType(SchemaConverters.java:232) ~[app.jar:?] 	at com.google.cloud.spark.bigquery.SchemaConverters.lambda$getDataType$4(SchemaConverters.java:182) ~[app.jar:?]
--
```

**Comment by davidrabinowitz** [comment]
Unfortunately it is not a simple conversion, as spark supports only decimal (38,38) which is not wide enough. Reducing the width may change the data when written back to BigQuery.

**Comment by richard-williamson** [comment]
@davidrabinowitz - thanks for the insight - just to get something moving we would be good with some simple casting - our initial use case is reading from bq to spark and the only reason we were using bignumeric instead of numeric for a couple tables was due to some data that had a large nbr of decimals for which we would have been fine to downcast intelligently as they appeared to be floating point errors/etc and in no way needed that precision (obviously we would want to error on the side of getting the integer part right first if there is room) - let me know if I can help provide any more initial requirements - spark users are typically fine with casting vs errors like this one :)

**Comment by Dee-Pac** [comment]
+1 for this issue. 
At PayPal - we have a lot of datasets defining BigNumeric type in BQ. Read uses-cases via spark are blocked.

**Comment by Dee-Pac** [comment]
@davidrabinowitz  Thanks for quick response.
- It may be possible to do what you've suggested above, especially when we are selecting specific columns.
- But use cases that require `select *` are the ones blocked. Although this is an anti-pattern to querying required columns in BQ - we have large scale data movement use cases. Under the hood - the frameworks applies `select *` and this is causing the pattern to break.

**Comment by sugyan84** [comment]
@himanshukohli09 Is this issue fixed and released yet ?

I'm using connector jar - gs://spark-lib/bigquery/spark-bigquery-latest_2.11.jar
and still see this issue. Please suggest.

**Comment by davidrabinowitz** [comment]
The latest version does not contain the bignumeric support yet, we will release it soon.

**Comment by davidrabinowitz** [comment]
Fixed in version 0.22.2

**Comment by KarthikeyanG98** [comment]
@davidrabinowitz can u pls post 0.22.0 link here.

**Comment by KarthikeyanG98** [comment]
> Fixed in version 0.22.0

can u pls post 0.22.0 link here.
